cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Hash.hs

```diff
@@ -16,21 +17,31 @@ import           Cardano.CLI.Types.Common
 
 import           Data.Text (Text)
 
-newtype GovernanceHashCmds era = GovernanceHashCmd (GovernanceHashCmdArgs era)
+data GovernanceHashCmds era
+  = GovernanceHashAnchorDataCmd !(GovernanceHashAnchorDataCmdArgs era)
+  | GovernanceHashScriptCmd !(GovernanceHashScriptCmdArgs era)
 
-data GovernanceHashCmdArgs era
-  = GovernanceHashCmdArgs {
+data GovernanceHashAnchorDataCmdArgs era
+  = GovernanceHashAnchorDataCmdArgs {
       eon     :: !(ConwayEraOnwards era)
-    , toHash  :: !GovernanceHashSource
+    , toHash  :: !GovernanceAnchorDataHashSource
     , moutFile :: !(Maybe (File () Out)) -- ^ The output file to which the hash is written
   } deriving Show
 
-data GovernanceHashSource
-  = GovernanceHashSourceBinaryFile (File ProposalText In)
-  | GovernanceHashSourceTextFile (File ProposalText In)
-  | GovernanceHashSourceText Text
+data GovernanceAnchorDataHashSource
+  = GovernanceAnchorDataHashSourceBinaryFile (File ProposalText In)
```

Comment:
`ProposalText` for binary files is quite misleading here. We should use use a different type here. Not necessarily within the scope of this PR.